lib/features/ticket/presentation/widgets/perk_card.dart

```diff
@@ -22,18 +24,50 @@ class PerkCard extends StatelessWidget {
   final Product product;
   final String price;
 
-  @override
-  Widget build(BuildContext context) {
-    return ShopCard(
-      title: title,
-      // TODO(fredpetersen): Get icon from product
-      icon: Icons.coffee_maker_outlined,
-      onTapped: (_) => buyModal(
+  void _onTapped(BuildContext context) {
+    if (product.price == 0 && product.amount == 1) {
+      showClaimSinglePerkConfirm(
+        context: context,
+        product: product,
+      );
+    } else if (product.price == 0 && product.amount > 1) {
+      appDialog(
```

Comment:
Shouldn't happen in prod, but claiming > 1 free product at a time is not intended behaviour. We show an error in this case